api-guidelines/v4.0/API-DESIGN-GUIDELINES/UNIFORM-RESOURCE-LOCATORS.md

```diff
@@ -0,0 +1,79 @@
+# Uniform Resource Locators (URLs)
+
+## URL Path Segments
+
+All URLs _must_ follow this pattern:
+
+```none
+[scheme]://[host](:[port])/[as-needed-path-segments]/[model namespace]/[resource name](/[identifier])(?[key=value])
+```
+
+* The scheme, host, and port are standard features of HTTP (examples:
+  `https://example.com` or `https://example.com:443`).
+* These _may_ optionally be followed by any path segments desired by the API
+  host. Example: `/api/data/v3`.
+* The _required_ model namespace is a feature of an Ed-Fi API; in the core Ed-Fi
+  Unifying Data Model, the namespace is always "ed-fi". Other values are used by
+  Extensions ot the Data Model.
+* The _required_ resource name corresponds to an entity in the Ed-Fi Unifying
+  Data Model. Examples of resource names, which are always plural: `students`,
+  `studentEducationOrganizationAssociations`.
```

Comment:
regarding plural, it might be obvious to most, but could be worth specifying that it follows standard english grammar and not "programmer plural" (localEducationAgencys vs localEducationAgencies). Not sure if anyone would actually do that, but it would make sense to me if I saw someone design such a thing